pkg/agent/secondarynetwork/init_test.go

```diff
@@ -0,0 +1,162 @@
+				m.EXPECT().Create().Return(nil)
+			},
+		},
+		{
+			name:        "create br error",
+			ovsBridges:  []string{"br1", "br2"},
+			expectedErr: "create error",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(ovsconfig.InvalidArgumentsError("create error"))
+			},
+		},
+		{
+			name:               "one interface",
+			ovsBridges:         []string{"br1"},
+			physicalInterfaces: []string{"eth1"},
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(nil)
+				m.EXPECT().GetOFPort("eth1", false).Return(int32(0), ovsconfig.InvalidArgumentsError("port not found"))
+				m.EXPECT().CreateUplinkPort("eth1", int32(0), map[string]interface{}{"antrea-type": "uplink"}).Return("", nil)
+			},
+		},
+		{
+			name:               "two interfaces",
+			ovsBridges:         []string{"br1", "br2"},
+			physicalInterfaces: []string{"eth1", "eth2"},
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(nil)
+				m.EXPECT().GetOFPort("eth1", false).Return(int32(0), ovsconfig.InvalidArgumentsError("port not found"))
+				m.EXPECT().CreateUplinkPort("eth1", int32(0), map[string]interface{}{"antrea-type": "uplink"}).Return("", nil)
+			},
+		},
+		{
+			name:               "interface already attached",
+			ovsBridges:         []string{"br1"},
+			physicalInterfaces: []string{"eth1"},
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(nil)
+				m.EXPECT().GetOFPort("eth1", false).Return(int32(0), nil)
+			},
+		},
+		{
+			name:               "non-existing interface",
+			ovsBridges:         []string{"br1"},
+			physicalInterfaces: []string{nonExistingInterface, "eth2"},
+			expectedErr:        "failed to get interface",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(nil)
+			},
+		},
+		{
+			name:               "create port error",
+			ovsBridges:         []string{"br1"},
+			physicalInterfaces: []string{"eth1"},
+			expectedErr:        "create error",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().Create().Return(nil)
+				m.EXPECT().GetOFPort("eth1", false).Return(int32(0), ovsconfig.InvalidArgumentsError("port not found"))
+				m.EXPECT().CreateUplinkPort("eth1", int32(0), map[string]interface{}{"antrea-type": "uplink"}).Return("", ovsconfig.InvalidArgumentsError("create error"))
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var bridges []agentconfig.OVSBridgeConfig
+			for _, brName := range tc.ovsBridges {
+				br := agentconfig.OVSBridgeConfig{BridgeName: brName}
+				br.PhysicalInterfaces = tc.physicalInterfaces
+				bridges = append(bridges, br)
+			}
+
+			controller := mock.NewController(t)
+			mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+
+			defer mockNewOVSBridge(mockOVSBridgeClient)()
+			defer mockInterfaceByName()()
```

Comment:
You can use the new `t.Cleanup` functionality here, to simplify calling the functions and ensure that cleanup always happens correctly. For example:
```go
func mockInterfaceByName(t *testing.T) func() {
	prevFunc := interfaceByNameFn
	interfaceByNameFn = func(name string) (*net.Interface, error) {
		if name == nonExistingInterface {
			return nil, errors.New("interface not found")
		}
		return nil, nil
	}
	t.Cleanup(func() { interfaceByNameFn = prevFunc })
}

// in caller
mockInterfaceByName(t)
```

Reply:
Done